src/payment/strategies/paypal-commerce/paypal-commerce-payment-processor.spec.ts

```diff
@@ -0,0 +1,62 @@
+import { MissingDataError, MissingDataErrorType } from '../../../common/error/errors';
+
+import { OverlayPaypal, PaypalCommercePaymentProcessor } from './index';
+
+// tslint:disable:no-non-null-assertion
+describe('PaypalCommercePaymentProcessor', () => {
+    let paypalCommercePaymentProcessor: PaypalCommercePaymentProcessor;
+    let overlay: OverlayPaypal;
+    const focus = jest.fn();
+    const close = jest.fn();
+
+    beforeEach(() => {
+        window.open = jest.fn(() => {
+            return { focus, close };
+        });
+
+        overlay = new OverlayPaypal();
+        paypalCommercePaymentProcessor = new PaypalCommercePaymentProcessor(overlay);
+    });
```

Comment:
The current implementation of `window.open` need only on the last `it` so, I recommend:

1)
```
beforeEach(() => {
    window.open = jest.fn();

    overlay = new OverlayPaypal();
    paypalCommercePaymentProcessor = new PaypalCommercePaymentProcessor(overlay);
});
```

2)
```
it('check event listener on message', async () => {
    window.open. mockImplementation(() => ({ focus, close }));
....
})
```

Reply:
Thank you!